src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private static final double DEFAULT_SIZE_GROW = 1.5;
+
+    private int size;
+    private T[] values;
+
+    public ArrayList() {
+        values = (T[]) new Object[MAX_ITEMS_NUMBER];
+    }
+
+    private void checkSize() {
+        if (size == values.length) {
+            values = getGrownArray();
+        }
+    }
+
+    private T[] getGrownArray() {
+        int newCapacity = 0;
+        int oldCapacity = values.length;
+        newCapacity = (int) (oldCapacity * DEFAULT_SIZE_GROW);
+
+        T[] newValues = (T[]) new Object[newCapacity];
+        System.arraycopy(values, 0, newValues, 0, size);
+        return newValues;
+    }
```

Comment:
1. Please place all private methods after all public ones
2. I'd not create separate method for 1 if statement. May be combined:
```suggestion
    private void checkSize() {
        if (size < values.length) {
            return;
        }
        int oldCapacity = values.length;
        int newCapacity = (int) (oldCapacity * DEFAULT_SIZE_GROW);

        T[] newValues = (T[]) new Object[newCapacity];
        System.arraycopy(values, 0, newValues, 0, size);
        values = newValues;
    }
```